release-notes/DIS/DIS_2.40.md

```diff
@@ -0,0 +1,90 @@
+| Plugins \> Add SNMP Trap Receiver... | Adds an SNMP trap receiver and a QAction with boilerplate code to process traps to the protocol. |
+| Plugins \> Add Table Context Menu... | Adds a custom context menu to a table in the protocol. You can choose between the following:<br>- Rows Manager (User-definable Keys): A default context menu that provides add, duplicate, edit and delete functionality.<br>- Rows Manager (Auto-incremented Keys): An extension of the previous type that allows developers to work with an auto-increment key parameter.<br>- Custom: Opens a wizard that allows you to specify the options. |
+
+#### XML editor: Specifying a project reference as DLL import [ID_34565]
+
+When, in the XML editor, you clicked the small *Down* arrow in front of a `<QAction>` tag and select *DLL Imports*, up to now, the submenu would list all QActions of which the `options` attribute contained the "precompile" option, as well as all commonly used system DLL files. From now on, it will also list QActions of which the C# project has a reference to another C# project in the solution that is not QAction_Helper, QAction_ClassLibrary or another QAction.
+
+Also, it is now possible to override the path associated with a DLL import. To do so, proceed as follows:
+
+1. Right-click the reference (which can be either a reference to a DLL or a reference to a project).
+1. Open its properties.
+1. Locate the *DLL Path* property, and enter in the value that should override the default path.
+
+#### DIS diagram: Third direction option "Both" [ID_34584]
+
+In the *DIS diagram* window, up to now, after selecting the required depth (i.e. the number of items you want to have displayed starting from or ending with the item you selected), you could select *Forward* or *Reverse* as direction. From now on, you can also select a third option: *Both*.
+
+| Direction | Description |
+|-----------|-------------|
+| Forward   | Shows the specified\* number of linked items starting from the item you selected. |
+| Reverse   | Shows the specified\* number of linked items ending with the item you selected.   |
+| Both      | Shows the specified\* number of linked items starting from the item you selected, as well as the specified\* number of linked items ending with the item you selected. |
+
+*\* The value entered in the Depth box.*
+
+#### XML editor: QAction\@dllImport attribute will now contain the full path to the DLL file [ID_34598]
+
+In the *QAction\@dllImport* attribute, you can now reference a DLL file that is stored in
+
+- a subfolder of the `Dlls` folder of the repository, or
+- a subfolder of the following `C:\Skyline DataMiner` folders: `Files`, `ProtocolScripts` or `ProtocolScripts\DllImport`
+
+Up to now, the *QAction\@dllImport* attribute would only contain the name of the DLL file. From now on, it will contain the full path to that file.
```

Comment:
```suggestion
Up to now, the *QAction\@dllImport* attribute would only contain the name of the DLL file. From now on, it will include the subfolders a well.
```